storage/remote/write_handler.go

```diff
@@ -62,7 +65,24 @@ func (h *writeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+func (h *writeHandler) checkAppendExemplarError(err error, e exemplar.Exemplar, outOfOrderExemplarErrs *int) error {
```

Comment:
Instead of replicating this function from the scrape loop we should move it somewhere both scrape and write_handler can use it.

Reply:
Could probably do the same with the actual calling of this function and the potential error message logging later on for lines 109-114 as well. I don't have a suggestion for where to move the code to resuse off the top of my head though.

Reply:
I was thinking of putting the shared code into `pkg/exemplar/storage.go`... WDYT?

Reply:
the function would look like this:
```golang
func CheckAppendExemplarError(logger log.Logger, err error, e Exemplar, exemplarOutOfOrder prometheus.Counter, outOfOrderExemplarErrs *int) error {
	switch errors.Cause(err) {
	case storage.ErrNotFound:
		return storage.ErrNotFound
	case storage.ErrOutOfOrderExemplar:
		*outOfOrderExemplarErrs++
		_ = level.Debug(logger).Log("msg", "Out of order exemplar", "exemplar", fmt.Sprintf("%+v", e))
		if exemplarOutOfOrder != nil {
			exemplarOutOfOrder.Inc()
		}
		return nil
	default:
		return err
	}
}
```

Reply:
Either `pkg/exemplar` or `storage/` makes sense to me, assuming you can avoid a circular import

Reply:
I had to put it in `storage/` since in `pkg/exemplar` it created a circular import